unstructured/ingest/connector/clarifai.py

```diff
@@ -0,0 +1,128 @@
+import os
+import typing as t
+from dataclasses import dataclass, field
+import uuid
+from pathlib import PurePath
+
+from unstructured.ingest.enhanced_dataclass import enhanced_field
+from unstructured.ingest.enhanced_dataclass.core import _asdict
+from unstructured.ingest.interfaces import (
+    AccessConfig,
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseSingleIngestDoc,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.ingest.error import DestinationConnectionError
+from unstructured.ingest.utils.data_prep import chunk_generator
+from unstructured.staging.base import flatten_dict
+from unstructured.utils import requires_dependencies
+
+if t.TYPE_CHECKING:
+    from clarifai.client.input import Inputs
+
+@dataclass
+class ClarifaiAccessConfig(AccessConfig):
+    api_key: str = enhanced_field(sensitive=True)
+    #print("class initialized")
+
+@dataclass
+class SimpleClarifaiConfig(BaseConnectorConfig):
+    access_config: ClarifaiAccessConfig
+    app_id: str
+    user_id: str
+    
+@dataclass
+class ClarifaiWriteConfig(WriteConfig):
+    batch_size: int = 50
+
+@dataclass
+class ClarifaiDestinationConnector(BaseDestinationConnector):
+    write_config: ClarifaiWriteConfig
+    connector_config: SimpleClarifaiConfig
+    _client: t.Optional["Inputs"] = field(init=False, default=None)
+    
+    @property
+    @requires_dependencies(["clarifai"], extras="clarifai")
+    def client(self) -> "Inputs":
+        if self._client is None:
+            from clarifai.client.input import Inputs
+        access_conf = self.connector_config.access_config
```

Comment:
Lets get line 51 through 60 indented under the `if`. Because if `self._client` is not None the method doesn't know anything about Inputs and throws an error.